File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/DeliveryStatusLabelsTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class DeliveryStatusLabelsTest {
  @Test
  fun `maps known delivery statuses to readable labels`() {
    assertEquals("Disponivel", deliveryStatusLabel("AGUARDANDO_MOTOBOY"))
    assertEquals("Aceita", deliveryStatusLabel("ACEITA_PELO_MOTOBOY"))
    assertEquals("Em rota", deliveryStatusLabel("EM_ROTA"))
  }

  @Test
  fun `formats unknown delivery statuses without raw underscores`() {
    assertEquals("Novo status", deliveryStatusLabel("NOVO_STATUS"))
  }

  @Test
  fun `uses a safe fallback for blank delivery statuses`() {
    assertEquals("Status desconhecido", deliveryStatusLabel("   "))
  }
}

File: apps/motoboy-pwa/src/deliveryStatusLabels.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import { deliveryStatusLabel } from "./deliveryStatusLabels";

describe("deliveryStatusLabel", () => {
  it("maps known delivery statuses to readable labels", () => {
    assert.equal(deliveryStatusLabel("AGUARDANDO_MOTOBOY"), "Disponivel");
    assert.equal(deliveryStatusLabel("ACEITA_PELO_MOTOBOY"), "Aceita");
    assert.equal(deliveryStatusLabel("EM_ROTA"), "Em rota");
  });

  it("formats unknown delivery statuses without raw underscores", () => {
    assert.equal(deliveryStatusLabel("NOVO_STATUS"), "Novo status");
  });

  it("uses a safe fallback for blank delivery statuses", () => {
    assert.equal(deliveryStatusLabel("   "), "Status desconhecido");
  });
});
